StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1108BlockStatementsMustNotContainEmbeddedComments.cs

```diff
@@ -58,19 +61,90 @@ public class SA1108BlockStatementsMustNotContainEmbeddedComments : DiagnosticAna
+            SyntaxKind.CatchClause,
+            SyntaxKind.FinallyClause,
+            SyntaxKind.CheckedStatement,
+            SyntaxKind.UncheckedStatement,
+            SyntaxKind.FixedStatement
+        };
+
+        /// <inheritdoc/>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => SupportedDiagnosticsValue;
+
         /// <inheritdoc/>
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(this.AnalyzeBlock, SyntaxKind.Block);
+            context.RegisterSyntaxNodeAction(this.AnalyzeSwitch, SyntaxKind.SwitchStatement);
+        }
+
+        private void AnalyzeSwitch(SyntaxNodeAnalysisContext context)
         {
-            get
+            var switchStatement = (SwitchStatementSyntax) context.Node;
+            var openBraceToken = switchStatement.OpenBraceToken;
+            if (openBraceToken.IsMissing)
             {
-                return SupportedDiagnosticsValue;
+                return;
             }
+
+            var previousToken = openBraceToken.GetPreviousToken();
+            if (previousToken.IsMissing)
+            {
+                return;
+            }
+
+            this.FindAllComments(context, previousToken, openBraceToken);
         }
 
-        /// <inheritdoc/>
-        public override void Initialize(AnalysisContext context)
+        private void AnalyzeBlock(SyntaxNodeAnalysisContext context)
+        {
+            var block = (BlockSyntax) context.Node;
+            if (block.Parent == null || !this.supportedKinds.Any(block.Parent.IsKind))
+            {
+                return;
+            }
+
+            var openBraceToken = block.OpenBraceToken;
+            if (openBraceToken.IsMissing)
+            {
+                return;
+            }
+
+            var previousToken = openBraceToken.GetPreviousToken();
+            if (previousToken.IsMissing)
+            {
+                return;
+            }
+
+            this.FindAllComments(context, previousToken, openBraceToken);
+        }
+
+        private void FindAllComments(SyntaxNodeAnalysisContext context, SyntaxToken previousToken, SyntaxToken openBraceToken)
+        {
+            var comments = previousToken.TrailingTrivia.Where(this.IsComment)
+                .Union(openBraceToken.LeadingTrivia.Where(this.IsComment))
+                .ToList();
+            foreach (var comment in comments)
+            {
+                context.ReportDiagnostic(Diagnostic.Create(Descriptor, comment.GetLocation()));
+            }
+        }
+
+
+        private bool IsComment(SyntaxTrivia syntaxTrivia)
         {
-            // TODO: Implement analysis
+            return (syntaxTrivia.IsKind(SyntaxKind.SingleLineCommentTrivia) && !syntaxTrivia.ToFullString().StartsWith(@"////")) ||
+                   syntaxTrivia.IsKind(SyntaxKind.MultiLineCommentTrivia);
```

Comment:
It's my preference too, but there's no standard for this issue right now.

:bulb: This would make a good proposal for a new Readability rule.